refiner/app/services/format.py

```diff
@@ -0,0 +1,34 @@
+from lxml import etree
+from lxml.etree import _Element
+
+
+def normalize_xml(xml: str) -> str:
+    """
+    Normalize XML string for comparison: parse, strip out all comments, then re-serialize with consistent pretty-printing.
+    """
+
+    # Reject non‐string inputs
+    if not isinstance(xml, str):
+        raise ValueError(f"Expected XML as str, got {type(xml).__name__!r}")
+
+    # Parse into an Element
+    root: _Element = etree.fromstring(xml)
+
+    # Force mypy to treat this as a list of Elements
+    comment_nodes: list[_Element] = root.xpath("//comment()")  # type: ignore[assignment]
```

Comment:
We use `cast` in a few other parts of the code base where do something similar. This should work for you here too:
```suggestion
    comment_nodes = cast(list[_Element], root.xpath("//comment()"))
```

We'll need to add `from typing import cast` the top of the file 🫡